kyo-core/shared/src/test/scala/kyoTest/KyoSpecAbstract.scala

```diff
@@ -0,0 +1,85 @@
+package kyoTest
+
+import kyo.*
+import kyo.KyoApp.Effects
+import scala.concurrent.Future
+import scala.concurrent.duration.Duration
+import scala.language.postfixOps
+import zio.EnvironmentTag
+import zio.Trace
+import zio.ZIO
+import zio.ZLayer
+import zio.internal.stacktracer.SourceLocation
+import zio.test.*
+
+abstract class KyoSpecAbstract[S] extends ZIOSpecAbstract:
```

Comment:
`ZIOSpecAbstract` has a timeout warning after 60 seconds. This works correctly with `Fiber.sleep`:

```
timestamp=2024-04-24T19:41:29.306520Z level=WARN thread=#zio-fiber-160508830 message="Test suite! - Fibers.sleep has taken more than 1 m to execute. If this is not expected, consider using TestAspect.timeout to timeout runaway tests for faster diagnostics."
```

Reply:
I think we need to propagate ZIO's cancelation to Kyo's fibers otherwise the test could continue running in the background.